### DISCUSSION TITLE
"This page isn't responding" When uploading many assets in backend

### DISCUSSION THREAD

**Discussion by IMCtoys** [pimcore/pimcore#16106] [Q&A / Support] [answered]
Hi,

we need to upload many assets, basically upload a folder structure from a windows folder to pimcore, but the browser always shows a popup saying that the browser is not responding. Then we click to "wait" and at the end it finishes. As I said we have a lot of images and folders and will take some time to upload everything, so this is a real problem for us. 

![image](https://github.com/pimcore/pimcore/assets/138874060/08710a90-8c25-42e2-942d-f70deaf4372d)

Is there any way to solve it? Alternatively, is there any way to upload assets from a backend process that won't rely on the browser?

thanks you!

**Comment by Maksonis** [comment, marked as answer]
Hi, you can use WebDAV access: https://pimcore.com/docs/platform/Pimcore/Assets/Accessing_Assets_via_WebDAV/. I used https://cyberduck.io/ as WebDAV client. Make sure you adapted your nginx configuration according to docs.

**Reply by IMCtoys** [reply]
Great, it seems a good solution! Thank you very much!
